source/base/Shmoose.cpp

```diff
@@ -251,10 +261,23 @@ void Shmoose::sendMessage(QString const &message, QString const &type)
 void Shmoose::sendFile(QString const &toJid, QString const &file)
 {
     bool shouldEncryptFile = lurchAdapter_->isOmemoUser(toJid) && (! settings_->getSendPlainText().contains(toJid));
+    Swift::JID receiverJid(toJid.toStdString());
+    Swift::IDGenerator idGenerator;
+    notSentMsgId_ = QString::fromStdString(idGenerator.generateID());
+
+    // messsage is added to the database 
+    persistence_->addMessage( notSentMsgId_,
+                          QString::fromStdString(receiverJid.toBare().toString()),
+                          QString::fromStdString(receiverJid.getResource()),
+                          file, QMimeDatabase().mimeTypeForFile(file).name(), 0, shouldEncryptFile ? 1 : 0);
 
```

Comment:
Move line 264 - 272 to MessageHandler. notSentMsgId_ can be the return value.

Reply:
I'm not sure to catch what you want. Moving it to MessageHandler means the entry won't be added in the database before sending the final send message, so it won't appear with its upload status in the UI and if upload takes time, UI won't show that upload is in progress.